Zend/tests/callable_self_parent_static_deprecation.phpt

```diff
@@ -9,27 +9,52 @@ class A {
 class B extends A {
     public function test() {
         // Different callables using self/parent/static
-        echo "Test different callables\n";
-        call_user_func("self::foo");
-        call_user_func("parent::foo");
-        call_user_func("static::foo");
-        call_user_func(["self", "foo"]);
-        call_user_func(["parent", "foo"]);
-        call_user_func(["static", "foo"]);
-        call_user_func(["B", "self::foo"]);
-        call_user_func(["B", "parent::foo"]);
-        call_user_func(["B", "static::foo"]);
-        call_user_func(["B", "A::foo"]);
+        $variants = [
+            '"self::foo"'            => "self::foo",
+            '"parent::foo"'          => "parent::foo",
+            '"static::foo"'          => "static::foo",
+            '["self", "foo"]'        => ["self", "foo"],
+            '["parent", "foo"]'      => ["parent", "foo"],
+            '["static", "foo"]'      => ["static", "foo"],
+            '["B", "self::foo"]'     => ["B", "self::foo"],
+            '["B", "parent::foo"]'   => ["B", "parent::foo"],
+            '["B", "static::foo"]'   => ["B", "static::foo"],
+            '["B", "A::foo"]'        => ["B", "A::foo"],
+            '[$this, "self::foo"]'   => [$this, "self::foo"],
+            '[$this, "parent::foo"]' => [$this, "parent::foo"],
+            '[$this, "static::foo"]' => [$this, "static::foo"],
+            '[$this, "A::foo"]'      => [$this, "A::foo"],
+        ];
+
+        echo "==> Test call_user_func\n";
+        foreach ( $variants as $description => $callable ) {
```

Comment:
Style nit:
```suggestion
        foreach ($variants as $description => $callable) {
```
Same below.

Reply:
Finally got round to fixing this; thanks for the review!